packages/fxa-content-server/app/scripts/lib/survey-filter.js

```diff
@@ -0,0 +1,243 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UserAgent from './user-agent';
+
+// All but the default export here is to make testing easier.
+
+const falseFn = () => false;
+export const NONE = Symbol();
+
+export const participatedRecently = (
+  previousParticipationTime,
+  doNotBotherSpan
+) => {
+  if (
+    previousParticipationTime === null ||
+    previousParticipationTime === undefined
+  ) {
+    return false;
+  }
```

Comment:
@meandavejustice Updated this and added a corresponding test at https://github.com/mozilla/fxa/pull/5405/files#diff-913ef57336afdd82b4c000a1aedc0056R801-R812